modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/maven/aether/KobaltMavenResolver.kt

```diff
@@ -38,6 +38,14 @@ class KobaltMavenResolver @Inject constructor(val settings: KobaltSettings,
     fun resolve(id: String, scope: Scope? = null, filter: DependencyFilter? = null): DependencyResult {
         val dependencyRequest = DependencyRequest(createCollectRequest(id, scope), filter)
         val result = system.resolveDependencies(session, dependencyRequest)
+        if (args.downloadSources) {
+            listOf("sources", "javadoc").forEach {
+                val artifact = DefaultArtifact(id)
+                val sourceArtifact = DefaultArtifact(artifact.groupId, artifact.artifactId, it, artifact.extension,
+                        artifact.version)
+                system.resolveDependencies(session, DependencyRequest(createCollectRequest(sourceArtifact, scope), filter))
```

Comment:
I just experimented with launching the server with `--downloadSources` and then synced, and it didn't go well. 

First of all, it downloads all javadoc/sources for all the dependencies (which is terrible) but worse, if one of these dependencies fails to resolve (e.g. javadoc missing, which happens quite a bit on JCenter), the whole process aborts.

So this is the wrong solution: the plug-in should never launch the server in `--downloadSources` mode.

Instead, let's implement the "Download sources" solution as I suggested. Maybe when the user presses this button, if you can't resolve this yourself, you can either

- Launch a second Kobalt process with `--downloadSources --resolve ...`
- Or if you don't want to do that, I can add an API call to do this and you just call the server with that API call to download the sources.

What do you think?